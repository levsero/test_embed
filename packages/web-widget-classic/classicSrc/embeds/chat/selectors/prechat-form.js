import { i18n } from 'classicSrc/app/webWidget/services/i18n'
import {
  getAccountDefaultDepartmentId,
  getDepartment,
  getDepartmentsList,
  getIsAuthenticated,
  getLoginSettings,
  getSocialLogin,
} from 'classicSrc/embeds/chat/selectors'
import {
  getDefaultFormFields,
  getPrechatFormFields,
  getPrechatFormSettings,
} from 'classicSrc/redux/modules/selectors'
import {
  getSettingsChatDepartment,
  getSettingsChatDepartmentsEnabled,
  getSettingsChatPrechatForm,
} from 'classicSrc/redux/modules/settings/settings-selectors'
import _ from 'lodash'
import { createSelector } from 'reselect'

const getCanViewDepartmentSelect = (state) => {
  const customerDefinedDepartmentsEnabled = getSettingsChatDepartmentsEnabled(state)
  const { departments = [] } = getPrechatFormFields(state)

  if (
    Array.isArray(customerDefinedDepartmentsEnabled) &&
    customerDefinedDepartmentsEnabled.length === 0
  ) {
    return false
  }

  if (departments.length === 0) {
    return false
  }

  if (departments.length === 1 && departments[0].isDefault) {
    return false
  }

  return true
}

const getDepartmentLabel = (state) => {
  const prechatFormSettings = getSettingsChatPrechatForm(state)
  const defaultFields = getDefaultFormFields(state)

  const labelFromSettings = prechatFormSettings?.departmentLabel
  const labelFromChat = defaultFields?.department?.label
  const fallbackLabel = i18n.t('embeddable_framework.chat.form.common.dropdown.chooseDepartment')

  return i18n.getSettingTranslation(labelFromSettings) || labelFromChat || fallbackLabel
}

export const getVisiblePrechatFields = (state, options = {}) => {
  const prechatFormFields = getPrechatFormFields(state)
  const loginSettings = getLoginSettings(state)
  const isAuthenticated = getIsAuthenticated(state)
  const { authenticated: isSociallyLoggedIn } = getSocialLogin(state)
  // If the user has selected an offline department,
  // then all other fields except the phone field are changed to be required
  const selectedDepartment = getDepartment(state, options.department)
  const isDepartmentOffline = selectedDepartment?.status === 'offline' ? true : undefined

  return [
    {
      id: 'name',
      title: i18n.t('embeddable_framework.common.textLabel.name'),
      required: Boolean(prechatFormFields.name?.required || isDepartmentOffline),
      visible: loginSettings.enabled && !isAuthenticated && !isSociallyLoggedIn,
      type: 'text',
    },
    {
      id: 'socialLogin',
      type: 'socialLogin',
      visible: loginSettings.enabled && !isAuthenticated && !isSociallyLoggedIn,
    },
    {
      id: 'email',
      title: i18n.t('embeddable_framework.common.textLabel.email'),
      required: Boolean(prechatFormFields.email?.required || isDepartmentOffline),
      visible: loginSettings.enabled && !isAuthenticated && !isSociallyLoggedIn,
      type: 'email',
    },
    {
      id: 'department',
      title: getDepartmentLabel(state),
      required: Boolean(prechatFormFields.department?.required),
      visible: getCanViewDepartmentSelect(state),
      type: 'dropdown',
      options: prechatFormFields.departments?.map((department) => ({
        name: department.name,
        value: department.id,
        disabled: department.disabled,
      })),
    },
    {
      id: 'phone',
      title: i18n.t('embeddable_framework.common.textLabel.phone_number'),
      required: Boolean(prechatFormFields.phone?.required),
      visible: loginSettings.enabled && !isAuthenticated && loginSettings.phoneEnabled,
      type: 'text',
    },
    {
      id: 'message',
      title: i18n.t('embeddable_framework.common.textLabel.message'),
      required: Boolean(prechatFormFields.message?.required || isDepartmentOffline),
      visible: true,
      type: 'textarea',
    },
  ].filter((field) => field.visible)
}

export const getPrechatGreeting = (state) => getPrechatFormSettings(state).message

export const getDefaultSelectedDepartment = createSelector(
  [getSettingsChatDepartment, getAccountDefaultDepartmentId, getDepartmentsList],
  (settingsDefaultDeparmentNameOrId, accountDefaultDepartmentId, departments) => {
    const departmentNameOrId = settingsDefaultDeparmentNameOrId || accountDefaultDepartmentId

    return _.find(
      departments,
      (dept) => dept.name.toLowerCase() === departmentNameOrId || dept.id === departmentNameOrId
    )
  }
)
