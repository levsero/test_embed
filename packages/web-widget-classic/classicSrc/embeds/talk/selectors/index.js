import { i18n } from 'classicSrc/app/webWidget/services/i18n'
import { CONTACT_OPTIONS } from 'classicSrc/embeds/talk/constants'
import {
  CALLBACK_ONLY,
  CALLBACK_AND_PHONE,
  CLICK_TO_CALL,
} from 'classicSrc/embeds/talk/talk-capability-types'
import { getSettingsTalkTitle } from 'classicSrc/redux/modules/settings/settings-selectors'
import _ from 'lodash'
import { createSelector } from 'reselect'
import isFeatureEnabled from '@zendesk/widget-shared-services/feature-flags'

export const getEmbeddedVoiceSupported = (_state) =>
  isFeatureEnabled(null, 'embedded_voice_enabled')

export const getIsCallInProgress = (state) => state.talk.embeddedVoiceCallStatus.isCallInProgress
export const getHasLastCallFailed = (state) => state.talk.embeddedVoiceCallStatus.hasLastCallFailed
export const getRecordingConsent = (state) => state.talk.recordingConsent
export const getUserRecordingConsentRequirement = (state) =>
  state.talk.embeddableConfig.recordingConsent
export const getMicrophoneMuted = (state) => state.talk.microphoneMuted
export const getTimeInCall = (state) => state.talk.timeInCall
export const getTalkEmbeddableConfig = (state) => state.talk.embeddableConfig
export const getPhoneNumber = createSelector(
  getTalkEmbeddableConfig,
  (config) => config.phoneNumber
)
export const getIsEmbeddedVoiceEnabled = (state) =>
  getTalkEmbeddableConfig(state)?.capability === CLICK_TO_CALL
export const getEmbeddableConfigEnabled = (state) => getTalkEmbeddableConfig(state).enabled
export const getDeferredStatusOnline = (state) =>
  getTalkEmbeddableConfig(state).deferredStatusOnline
export const getTalkEmbeddableConfigConnected = (state) => getTalkEmbeddableConfig(state).connected
export const getAgentAvailability = (state) => state.talk.agentAvailability
export const getFormState = (state) => state.talk.formState
export const getCallback = (state) => state.talk.callback
export const getAverageWaitTime = (state) => state.talk.averageWaitTime.waitTime
export const getAverageWaitTimeEnabled = (state) => state.talk.averageWaitTime.enabled
export const getSocketIoVendor = (state) => state.talk.vendor.io
export const getIsPollingTalk = (state) => state.talk.isPolling

export const isCallbackEnabled = (state) => {
  const { capability } = state.talk.embeddableConfig

  return _.includes([CALLBACK_ONLY, CALLBACK_AND_PHONE], capability)
}

export const getAverageWaitTimeString = createSelector(
  [getAverageWaitTimeEnabled, getAverageWaitTime],
  (averageWaitTimeEnabled, averageWaitTime) => {
    const averageWaitTimeNumber = parseInt(averageWaitTime, 10)

    if (!averageWaitTimeEnabled || averageWaitTimeNumber === 0) return null

    const waitTimeForm = averageWaitTimeNumber > 1 ? 'Plural' : 'Singular'

    return i18n.t(`embeddable_framework.talk.form.averageWaitTime${waitTimeForm}`, {
      averageWaitTime,
    })
  }
)

export const getCapability = createSelector(
  [getTalkEmbeddableConfig, getEmbeddedVoiceSupported],
  (talkConfig, embeddedVoiceSupported) => {
    return embeddedVoiceSupported ? CLICK_TO_CALL : talkConfig.capability
  }
)

export const getOfflineTitle = (state) => {
  const capability = getCapability(state)
  const title = i18n.getSettingTranslation(getSettingsTalkTitle(state))

  if (title) {
    return title
  }

  switch (capability) {
    case CONTACT_OPTIONS.PHONE_ONLY:
      return i18n.t('embeddable_framework.talk.phoneOnly.title')
    case CONTACT_OPTIONS.CLICK_TO_CALL:
      return i18n.t('embeddable_framework.talk.embeddedVoice.channel.title')
    case CONTACT_OPTIONS.CALLBACK_AND_PHONE:
    case CONTACT_OPTIONS.CALLBACK_ONLY:
    default:
      return i18n.t('embeddable_framework.talk.form.title')
  }
}

export const getTalkTitleKey = createSelector(
  [getCapability, isCallbackEnabled],
  (capability, callbackEnabled) => {
    if (capability === CLICK_TO_CALL) return 'embeddable_framework.talk.embeddedVoice.channel.title'
    if (callbackEnabled) {
      return 'embeddable_framework.launcher.label.talk.request_callback'
    } else {
      return 'embeddable_framework.launcher.label.talk.call_us'
    }
  }
)
