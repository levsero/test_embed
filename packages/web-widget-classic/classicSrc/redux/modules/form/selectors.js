export const getFormValues = (state, formId) => state.form.formValues[formId]
