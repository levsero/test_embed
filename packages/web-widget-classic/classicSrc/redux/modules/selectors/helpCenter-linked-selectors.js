import {
  getIsShowHCIntroState,
  getHelpCenterSignInRequired,
} from 'classicSrc/embeds/helpCenter/selectors'
import {
  getActiveEmbed,
  getHelpCenterEmbed,
  getBaseIsAuthenticated,
} from 'classicSrc/redux/modules/base/base-selectors'
import { getSettingsHelpCenterSuppress } from 'classicSrc/redux/modules/settings/settings-selectors'
import { createSelector } from 'reselect'
import { isMobileBrowser, isOnHelpCenterPage } from '@zendesk/widget-shared-services'

export const getHelpCenterEnabled = createSelector(
  [getHelpCenterEmbed, getSettingsHelpCenterSuppress],
  (helpCenterEmbed, suppress) => {
    return helpCenterEmbed && !suppress
  }
)

export const getHasPassedAuth = createSelector(
  [getBaseIsAuthenticated, getHelpCenterSignInRequired, isOnHelpCenterPage],
  (isAuthenticated, helpCenterSignInRequired, isOnHelpCenterPage) => {
    return isAuthenticated || !helpCenterSignInRequired || isOnHelpCenterPage
  }
)

export const getHelpCenterAvailable = createSelector(
  [getHelpCenterEnabled, getHasPassedAuth],
  (helpCenterEnabled, hasPassedAuth) => {
    return helpCenterEnabled && hasPassedAuth
  }
)

export const getCanShowHelpCenterIntroState = createSelector(
  [getIsShowHCIntroState, getHelpCenterAvailable, getActiveEmbed],
  (isShowHCIntroState, isHelpCenterAvailable, activeEmbed) => {
    return (
      !isMobileBrowser() &&
      isShowHCIntroState &&
      isHelpCenterAvailable &&
      activeEmbed === 'helpCenterForm'
    )
  }
)

export const getHelpCenterReady = (state) => !getHelpCenterEmbed(state) || getHasPassedAuth(state)
