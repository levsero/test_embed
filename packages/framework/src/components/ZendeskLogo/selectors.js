import { getZopimId, getLocale } from 'src/redux/modules/base/base-selectors'
import { document, getZendeskHost } from 'utility/globals'
import { i18n } from 'service/i18n'

export const getZendeskLogoLink = (state, linkToChat) => {
  const zopimId = getZopimId(state)
  const locale = getLocale(state)

  if (linkToChat) {
    const params = [
      '?utm_source=webwidgetchat',
      '&utm_medium=poweredbyzendeskchat',
      '&utm_campaign=poweredbyzendesk',
      `&utm_term=${zopimId}`,
      `&utm_content=${getZendeskHost(document)}`,
      `&iref=${zopimId}`,
      `&lang=${locale}`
    ].join('')

    return [i18n.t('embeddable_framework.zendeskLogo.powered_by_url.chat'), params].join('')
  }

  return [
    i18n.t('embeddable_framework.zendeskLogo.powered_by_url.embeddables'),
    '?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image'
  ].join('')
}
