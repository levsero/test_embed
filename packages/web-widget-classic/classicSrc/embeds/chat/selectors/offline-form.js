import { i18n } from 'classicSrc/app/webWidget/services/i18n'
import {
  getIsAuthenticated,
  getLoginSettings,
  getSocialLogin,
} from 'classicSrc/embeds/chat/selectors'
import { getOfflineFormFields } from 'classicSrc/redux/modules/selectors'

export const getFields = (state) => {
  const { phoneEnabled } = getLoginSettings(state)
  const offlineFormFields = getOfflineFormFields(state)
  const isAuthenticated = getIsAuthenticated(state)
  const sociallyLoggedIn = getSocialLogin(state).authenticated

  return [
    {
      id: 'name',
      title: i18n.t('embeddable_framework.common.textLabel.name'),
      required: Boolean(offlineFormFields.name?.required),
      visible: !isAuthenticated && !sociallyLoggedIn,
      type: 'text',
    },
    {
      id: 'socialLogin',
      type: 'socialLogin',
      visible: !isAuthenticated && !sociallyLoggedIn,
    },
    {
      id: 'email',
      title: i18n.t('embeddable_framework.common.textLabel.email'),
      required: Boolean(offlineFormFields.email?.required),
      visible: !isAuthenticated && !sociallyLoggedIn,
      type: 'text',
    },
    {
      id: 'phone',
      title: i18n.t('embeddable_framework.common.textLabel.phone_number'),
      required: Boolean(offlineFormFields.phone?.required),
      visible: !isAuthenticated && phoneEnabled,
      type: 'text',
    },
    {
      id: 'message',
      title: i18n.t('embeddable_framework.common.textLabel.message'),
      required: Boolean(offlineFormFields.message?.required),
      visible: true,
      type: 'textarea',
    },
  ].filter((field) => field.visible)
}
