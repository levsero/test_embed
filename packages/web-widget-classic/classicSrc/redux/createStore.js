import { trackAnalytics } from 'classicSrc/redux/middleware/analytics'
import { sendBlips } from 'classicSrc/redux/middleware/blip'
import onStateChangeFn from 'classicSrc/redux/middleware/onStateChange/onStateChange'
import onStateChangeWrapper from 'classicSrc/redux/middleware/onStateChange/wrapper'
import persist from 'classicSrc/redux/middleware/persist'
import preventLoops from 'classicSrc/redux/middleware/preventLoops'
import queueCalls from 'classicSrc/redux/middleware/queue'
import resetActiveEmbed from 'classicSrc/redux/middleware/resetActiveEmbed'
import throttle from 'classicSrc/redux/middleware/throttle'
import reducer from 'classicSrc/redux/modules/reducer'
import { createStore, compose, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'
import { inDebugMode } from '@zendesk/widget-shared-services'

const reduxConsoleLogger = (storeName) => {
  return createLogger({
    collapsed: true,
    titleFormatter: (action) => [`${storeName}`, `%c${String(action.type)}`, '%c'].join(' '),
  })
}

const reduxDevtoolsAvailable = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
const reduxDevtoolsComposer = (storeName) => {
  return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ name: storeName })
}

export default function (storeName = 'web_widget', options = {}) {
  const middlewares = [
    thunk,
    preventLoops,
    throttle(options.throttleEvents, options.allowedActionsFn),
    onStateChangeWrapper(onStateChangeFn),
    sendBlips,
    onStateChangeWrapper(resetActiveEmbed),
    trackAnalytics,
    persist,
    onStateChangeWrapper(queueCalls),
  ]

  let composeEnhancers = compose

  if (inDebugMode()) {
    if (reduxDevtoolsAvailable) {
      composeEnhancers = reduxDevtoolsComposer(storeName)
    } else {
      middlewares.push(reduxConsoleLogger(storeName))
    }
  }

  return composeEnhancers(applyMiddleware(...middlewares))(createStore)(reducer)
}
