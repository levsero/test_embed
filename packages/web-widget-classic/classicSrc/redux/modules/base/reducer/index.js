import { combineReducers } from 'redux'
import backButtonVisible from './back-button-visibility'
import activeEmbed from './base-active-embed'
import afterWidgetShowAnimation from './base-after-widget-show-animation'
import arturos from './base-arturos'
import bootupTimeout from './base-bootupTimeout'
import widgetShown from './base-embed-shown'
import embeddableConfig from './base-embeddable-config'
import embeds from './base-embeds'
import hasWidgetShown from './base-has-widget-shown'
import hidden from './base-hidden'
import isAuthenticationPending from './base-is-authentication-pending'
import isChatBadgeMinimized from './base-is-chat-badge-minimized'
import locale from './base-locale'
import queue from './base-queue'
import widgetInitialised from './base-widget-initialised'
import launcherVisible from './launcher-visibility'
import webWidgetOpen from './web-widget-open'

export default combineReducers({
  activeEmbed,
  afterWidgetShowAnimation,
  arturos,
  backButtonVisible,
  bootupTimeout,
  embeddableConfig,
  embeds,
  hasWidgetShown,
  hidden,
  isAuthenticationPending,
  isChatBadgeMinimized,
  launcherVisible,
  locale,
  queue,
  widgetInitialised,
  webWidgetOpen,
  widgetShown,
})
