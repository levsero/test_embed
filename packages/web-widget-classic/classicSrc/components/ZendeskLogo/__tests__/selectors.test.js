import createStore from 'classicSrc/redux/createStore'
import { LOCALE_SET } from 'classicSrc/redux/modules/base/base-action-types'
import { updateEmbeddableConfig } from 'classicSrc/redux/modules/base/base-actions'
import { getZendeskLogoLink } from '../selectors'

describe('getZendeskLogoLink', () => {
  it('return chat url if on chat product', () => {
    const mockEmbeddableConfig = {
      embeds: {
        chat: {
          props: {
            zopimId: 'randomZopimId',
          },
        },
      },
    }

    const store = createStore()

    store.dispatch({
      type: LOCALE_SET,
      payload: 'en-US',
    })
    store.dispatch(updateEmbeddableConfig(mockEmbeddableConfig))
    const url = getZendeskLogoLink(store.getState(), true)

    expect(url).toMatchSnapshot()
  })

  it('return generic web widget url if not on a chat product', () => {
    const store = createStore()

    store.dispatch({
      type: LOCALE_SET,
      payload: 'en-US',
    })
    const url = getZendeskLogoLink(store.getState(), false)

    expect(url).toMatchSnapshot()
  })
})
