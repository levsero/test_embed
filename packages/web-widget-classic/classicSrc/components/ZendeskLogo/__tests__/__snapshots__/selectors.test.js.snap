// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getZendeskLogoLink return chat url if on chat product 1`] = `"https://www.zendesk.com/chat/live-chat-widget/?utm_source=webwidgetchat&utm_medium=poweredbyzendeskchat&utm_campaign=poweredbyzendesk&utm_term=randomZopimId&utm_content=testingHost&iref=randomZopimId&lang=en-US"`;

exports[`getZendeskLogoLink return generic web widget url if not on a chat product 1`] = `"https://www.zendesk.com/embeddables/?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"`;
